src/pytorch_lightning/strategies/hpu_deepspeed.py

```diff
@@ -0,0 +1,74 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import torch
+
+import pytorch_lightning as pl
+from pytorch_lightning.plugins.environments.cluster_environment import ClusterEnvironment
+from pytorch_lightning.plugins.precision import PrecisionPlugin
+from pytorch_lightning.strategies.hpu_parallel import HPUParallelStrategy
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+
+
+class HPUDeepSpeedStrategy(HPUParallelStrategy):
```

Comment:
We shouldn't be dependent on DDP based strategy for deep speed

Reply:
@jerome-habana why not?

Reply:
generally DS implements the reductions differently unlike DDP.

Reply:
doing the overwriting